dataplane/src/main.rs

```diff
@@ -152,14 +152,6 @@ fn main() {
 mod test {
     use n_vm::in_vm;
 
-    #[allow(unreachable_code, clippy::should_panic_without_expect)] // demo test to check vm functionality
-    #[should_panic]
-    #[test]
-    #[in_vm]
-    fn panic_in_vm_shows_as_test_failure() {
-        panic!("this panic should be caught by the test framework and show as a test failure");
-    }
```

Comment:
Would it be worth to keep but comment this test (with an additional comment explaining the race we've been observing)?